homeassistant/components/wyoming/__init__.py

```diff
@@ -25,23 +33,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if service is None:
         raise ConfigEntryNotReady("Unable to connect")
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = service
+    satellite_devices = SatelliteDevices(hass, entry)
+    satellite_devices.async_setup()
+
+    item = DomainDataItem(service=service, satellite_devices=satellite_devices)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = item
 
     await hass.config_entries.async_forward_entry_setups(
         entry,
-        service.platforms,
+        service.platforms + PLATFORMS,
     )
 
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    if service.info.satellite is not None:
+        satellite_device = satellite_devices.async_get_or_create(item.service)
+        wyoming_satellite = WyomingSatellite(hass, service, satellite_device)
+        hass.async_create_background_task(
+            wyoming_satellite.run(), f"Satellite {item.service.info.satellite.name}"
+        )
+
+        def stop_satellite():
+            wyoming_satellite.is_running = False
```

Comment:
Is this enough or should you also call a function to send a signal to the run loop?

Reply:
The `is_running` property was sending that signal, but I'm refactoring it into a proper `stop()` method to make it more clear.